python/pyspark/sql/dataframe.py

```diff
@@ -2020,8 +2021,6 @@ def _to_corrected_pandas_type(dt):
         return np.int32
     elif type(dt) == FloatType:
         return np.float32
-    elif type(dt) == DateType:
-        return 'datetime64[ns]'
```

Comment:
I thought we were considering the interpretation of DateType as object as a bug, similar to how FloatType was being interpreted as float64?

Reply:
+1, I feel it was a bug. Maybe we can merge this to branch-2.3 only and update the migration guide in the master branch?